src/csharp/Grpc.HealthCheck/HealthGrpc.cs

```diff
@@ -38,11 +38,22 @@ public interface IHealthClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IHealth
     {
       Task<global::Grpc.Health.V1.HealthCheckResponse> Check(global::Grpc.Health.V1.HealthCheckRequest request, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class HealthBase
```

Comment:
It feels like making the new class implement an interface that is going away soon isn't a good idea.
How exactly would "class HealthBase : IHealth" be easier for people to transition? Please note that I now generate two overloads for the method BindService - `BindService(IBenchmarkService serviceImpl)` and `BindService(BenchmarkServiceBase serviceImpl)`.